calico_cni_k8s_test.go

```diff
@@ -396,6 +396,95 @@ var _ = Describe("CalicoCni", func() {
 					}))
 				})
 			})
+
+			Context("Using host-local IPAM: request an IP then release it, and then request it again", func() {
+				It("Should successfully assign IP both times and successfully release it in the middle", func() {
+					netconfHostLocalIPAM := fmt.Sprintf(`
+				  {
+					"name": "net5",
+					  "type": "calico",
+					  "etcd_endpoints": "http://%s:2379",
+					  "ipam": {
+					    "type": "host-local",
+						"subnet": "usePodCidr"
+					  },
+				   "kubernetes": {
+				     "k8s_api_root": "http://127.0.0.1:8080"
+			    	},
+			   	  "policy": {"type": "k8s"},
+				  "log_level":"info"
+					}`, os.Getenv("ETCD_IP"))
+
+					config, err := clientcmd.DefaultClientConfig.ClientConfig()
+					Expect(err).NotTo(HaveOccurred())
+
+					clientset, err := kubernetes.NewForConfig(config)
+					Expect(err).NotTo(HaveOccurred())
+
+					// Create a K8s Node object with PodCIDR and name equal to hostname.
+					_, err = clientset.Nodes().Create(&v1.Node{
+						ObjectMeta: v1.ObjectMeta{Name: hostname},
+						Spec: v1.NodeSpec{
+							PodCIDR: "10.0.0.0/24",
+						},
+					})
+					Expect(err).NotTo(HaveOccurred())
+
+					// Step 1: Create the first pod on node we created.
+					name := fmt.Sprintf("run%d", rand.Uint32())
+
+					// Create a K8s pod with the node we created above.
+					_, err = clientset.Pods(K8S_TEST_NS).Create(&v1.Pod{
+						ObjectMeta: v1.ObjectMeta{Name: name},
+						Spec: v1.PodSpec{Containers: []v1.Container{{
+							Name:  fmt.Sprintf("container-%s", name),
+							Image: "ignore",
+						}},
+							NodeName: hostname,
+						},
+					})
+					Expect(err).NotTo(HaveOccurred())
+
+					// host-local IPAM always assigns the IPs sequentialls except the first 2 that are reserved
+					expectedIP := net.IPv4(10, 0, 0, 2).To4()
+
+					_, netnspath, session, _, contAddresses, _, err := CreateContainer(netconfHostLocalIPAM, name)
+					Expect(err).NotTo(HaveOccurred())
+
+					podIP := contAddresses[0].IP
+					logger.Infof("All container IPs: %v", contAddresses)
+					logger.Infof("Container got IP address: %s", podIP)
+					Expect(podIP).Should(Equal(expectedIP))
+
+					// Step 2: Delete the pod we created earlier, so it should release the IP address.
+					session, err = DeleteContainer(netconfHostLocalIPAM, netnspath, name)
+					Expect(err).ShouldNot(HaveOccurred())
+					Eventually(session).Should(gexec.Exit())
+
+					// Step 3: Create a second pod on the same node, it should get the same IP first pod got.
+					name2 := fmt.Sprintf("run2%d", rand.Uint32())
+
+					// Create another K8s pod with the node we created above.
```

Comment:
Same here w.r.t the comments.  This is just a duplicate of the "Step 3:" comment.